runtime/src/qio/auxFilesys/hdfs/qio_plugin_hdfs.c

```diff
@@ -493,10 +493,15 @@ const qio_file_functions_ptr_t hdfs_function_struct_ptr = &hdfs_function_struct;
 static
 char* get_locale_name(char *loc)
 {
+  char* ret = NULL;
   const char *keys = "."; // We lop off the .us.cray...
-  int i = strcspn (loc, keys);
-
-  return strndup(loc, i);
+  int len = strcspn (loc, keys);
+  ret = (char*) qio_malloc(len+1);
+  if (ret) {
+    ret[len] = '\0';
+    qio_memcpy(ret, loc, len);
```

Comment:
Looks good. I'd tend to write the '\0' after the memcpy (easier to optimize), but that's a minor issue & performance is not critical here.